.github/workflows/remove_deprecated_decls.yml

```diff
@@ -162,21 +164,62 @@ jobs:
         echo "Deleting ${tmplean}"
         rm "${tmplean}"
 
+    - name: Get branch SHA if it exists
+      id: get-branch-sha
+      run: |
+        # Check if the branch exists remotely
+        if git fetch origin "$BRANCH_NAME"; then
+          SHA=$(git rev-parse "origin/$BRANCH_NAME")
+          echo "Branch '$BRANCH_NAME' exists with SHA: $SHA"
+          echo "sha=$SHA" >> "${GITHUB_OUTPUT}"
+        else
+          echo "Branch '$BRANCH_NAME' does not exist"
+          echo "sha=" >>" ${GITHUB_OUTPUT}"
+        fi
+
+    - name: Get PR
+      if: ${{ steps.get-branch-sha.outputs.sha != '' }}
+      id: find-pr # all the steps below are skipped if 'ready-to-merge' is in the list of labels found here
+      uses: 8BitJonny/gh-get-current-pr@4056877062a1f3b624d5d4c2bedefa9cf51435c9 # 4.0.0
+      # TODO: this may not work properly if the same commit is pushed to multiple branches:
+      # https://github.com/8BitJonny/gh-get-current-pr/issues/8
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        sha: ${{ steps.get-branch-sha.outputs.sha }}
+        # Only return if PR is still open
+        filterOutClosed: true
+
+    - name: Send Zulip message (blocked by open PR)
+      if: steps.find-pr.outputs.pr_found == 'true'
+      uses: zulip/github-actions-zulip/send-message@e4c8f27c732ba9bd98ac6be0583096dea82feea5 # v1.0.2
+      with:
+        api-key: ${{ secrets.ZULIP_API_KEY }}
+        email: 'github-mathlib4-bot@leanprover.zulipchat.com'
+        organization-url: 'https://leanprover.zulipchat.com'
+        to: 'nightly-testing'
+        type: 'stream'
+        topic: 'Mathlib `remove outdated deprecated declarations`'
+        content: |
+          Reminder: #${{ steps.find-pr.outputs.pr }} is still open; this has prevented the automatic
+          deprecation workflow from creating a new PR.
```

Comment:
Just one more comment: what do you think of linking also the job number that created this message, so that it can be easily seen what happened?